crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -59,287 +59,269 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
         let source = f.context().source();
         let source_type = f.options().source_type();
 
-        let mut f = WithNodeLevel::new(node_level, f);
-        write!(
-            f,
-            [format_with(|f| {
-                let mut iter = statements.iter();
-                let Some(first) = iter.next() else {
-                    return Ok(());
-                };
-
-                // Format the first statement in the body, which often has special formatting rules.
-                let first = match self.kind {
-                    SuiteKind::Other => {
-                        if is_class_or_function_definition(first) && !comments.has_leading(first) {
-                            // Add an empty line for any nested functions or classes defined within
-                            // non-function or class compound statements, e.g., this is stable formatting:
-                            // ```python
-                            // if True:
-                            //
-                            //     def test():
-                            //         ...
-                            // ```
-                            empty_line().fmt(f)?;
-                        }
-
-                        SuiteChildStatement::Other(first)
-                    }
+        let mut with_node_level = WithNodeLevel::new(node_level, f);
```

Comment:
My reasoning for using `format_with` is that `Formatter::write_fmt` (which `write` calls) is a static dispatch. Whereas `X.write_fmt` requires creating a new `Formatter`. But it probably doesn't matter and is just noise.

Reply:
but we don't call `write_fmt`, do we?

Reply:
Ah, this works because `WithNodeLevel` is generic over `Buffer`